editor/components/post-author/index.js

```diff
@@ -5,39 +5,86 @@ import { __ } from '@wordpress/i18n';
 import { withInstanceId } from '@wordpress/components';
 import { Component, compose } from '@wordpress/element';
 import { withSelect, withDispatch } from '@wordpress/data';
+import apiRequest from '@wordpress/api-request';
 
 /**
  * Internal dependencies
  */
 import PostAuthorCheck from './check';
 
+/**
+ * External dependencies
+ */
+import accessibleAutocomplete from 'accessible-autocomplete';
+import './accessible-autocomplete.css';
+import { findWhere, debounce } from 'underscore';
+
 export class PostAuthor extends Component {
 	constructor() {
 		super( ...arguments );
 
 		this.setAuthorId = this.setAuthorId.bind( this );
+		this.suggestAuthor = this.suggestAuthor.bind( this );
+	}
+
+	componentDidMount() {
```

Comment:
You could [import a React component version of the autocomplete](https://github.com/alphagov/accessible-autocomplete#using-with-react) which would have two main benefits; one of avoiding a clunky `componentDidMount` + `document.querySelector` and also of sharing React as the peer dependency used in the rest of gutenberg, which would reduce the bundle size by ~3kb. I think you would have to reimplement part of [enhanceSelectElement](https://github.com/alphagov/accessible-autocomplete/blob/master/src/wrapper.js#L19) to fit in with the existing elements in this component. I recommend sticking to server-side rendering the select and then switching to `<Autocomplete />`, possibly on `componentDidMount`.

Reply:
@tvararu thanks for the feedback. I did try using the built in react component and ran into some issues that led me to the current approach. I didn't fully consider the bundle size and agree this might be worth exploring further.

Reply:
@adamsilverstein I've successfully used the React component in a Gutenberg block in the past. Here's the code in case it might be helpful: https://gist.github.com/swissspidy/86861f9012d96298f735987388c0fd23

Reply:
thanks @swissspidy i'll give this another pass to see if i can get it working correctly with the React component and report back with any issues I encounter.

Reply:
Clarifying: this comment is outdated: the current version uses the React component: https://github.com/WordPress/gutenberg/pull/7385/files#diff-8267665c4a351e7321fa6763c14922d3R18